aea/configurations/validation.py

```diff
@@ -313,3 +313,31 @@ def check_excludes(path):
             )
 
     return errors
+
+
+SAME_MARK = object()
+
+
+def filter_data(base: Any, updates: Any) -> Any:
```

Comment:
Can we use type `JSONLike` instead (from `aea.common`)?

Reply:
there is not only json friendly data, as i remember component_id also can be provided